src/pages/groupComparison/GroupComparisonMutationTable.tsx

```diff
@@ -0,0 +1,347 @@
+                        <td>: Significant association</td>
+                    </tr>
+                </table>
+            ),
+        };
+
+        this._columns[MutationTableColumnType.MUTATION_OVERLAP] = {
+            name: MutationTableColumnType.MUTATION_OVERLAP,
+            headerRender: () => <span>Co-occurrence Pattern</span>,
+            render: (d: Mutation[]) =>
+                mutationOverlapRenderFunction(
+                    this.props.rowDataByProteinChange,
+                    d,
+                    this.props.profiledPatientCounts,
+                    this.props.sampleSet,
+                    this.props.groups
+                ),
+            tooltip: (
+                <table>
+                    <tr>
+                        <td>Upper row</td>
+                        <td>: Patients colored according to group.</td>
+                    </tr>
+                    <tr>
+                        <td>Lower row</td>
+                        <td>
+                            : Patients with a mutation in the selected gene for
+                            the listed protein change are highlighted.
+                        </td>
+                    </tr>
+                </table>
+            ),
+        };
+
+        this._columns[MutationTableColumnType.P_VALUE] = {
+            name: 'p-Value',
+            render: (d: Mutation[]) =>
+                pValueRenderFunction(this.props.rowDataByProteinChange, d),
+            download: (d: Mutation[]) =>
+                getPValueTextValue(this.props.rowDataByProteinChange, d),
+            sortBy: (d: Mutation[]) =>
+                getPValueData(this.props.rowDataByProteinChange, d),
+            tooltip: <span>Derived from two-sided Fisher Exact test</span>,
+        };
+
+        this._columns[MutationTableColumnType.Q_VALUE] = {
+            name: 'q-Value',
+            render: (d: Mutation[]) =>
+                qValueRenderFunction(this.props.rowDataByProteinChange, d),
+            download: (d: Mutation[]) =>
+                getQValueTextValue(this.props.rowDataByProteinChange, d),
+            sortBy: (d: Mutation[]) =>
+                getQValueData(this.props.rowDataByProteinChange, d),
+            tooltip: <span>Derived from Benjamini-Hochberg procedure</span>,
+        };
+
+        // comparison annotation column render
+        this._columns[MutationTableColumnType.ANNOTATION].render = (
+            d: Mutation[]
+        ) => (
+            <span id="mutation-annotation">
+                {AnnotationColumnFormatter.renderFunction(
+                    d,
+                    {
+                        hotspotData: this.props.hotspotData,
+                        myCancerGenomeData: this.props.myCancerGenomeData,
+                        oncoKbData: haveDifferentCancerTypes(
+                            d,
+                            this.resolveTumorType
+                        )
```

Comment:
We have redundant calls to the function `haveDifferentCancerTypes`. Instead we should just call it once, at the very beginning of the function. Something like

```JavaScript
const hasMultipleCancerTypes = haveDifferentCancerTypes(...)
```